src/nodes.coffee

```diff
@@ -2209,7 +2209,7 @@ exports.Assign = class Assign extends Base
     answer = compiledName.concat @makeCode(" #{ @context or '=' } "), val
     # Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration,
     # if we’re destructuring without declaring, the destructuring assignment must be wrapped in parentheses.
-    if o.level > LEVEL_LIST or o.level is LEVEL_TOP and isValue and @variable.base instanceof Obj and not @nestedLhs and not (@param is yes)
+    if o.level > LEVEL_LIST or (o.level is LEVEL_TOP or o.level is LEVEL_LIST) and isValue and @variable.base instanceof Obj and not @nestedLhs and not (@param is yes)
```

Comment:
The first > could be >=?

Reply:
That sounds like an improvement, yes.

Side note: Can someone who really understands `LEVEL_TOP`, `LEVEL_LIST` etc. write up an explanation on one of the wiki pages?

Reply:
@vendethiel, @GeoffreyBooth 
The condition has basically just left and right side, and could be written as

```coffeescript
if o.level > LEVEL_LIST or (o.level is LEVEL_TOP and isLevel and ...)
```

So, if we change first `>` to `>=`, the left side condition will be also true in cases like
` a = (b=1) -> `, where the level of assignment `b=5` is `LEVEL_LIST`, and will be compiled to  
`a = function((b=1)) {}`.

However, the condition could be simplified a bit and changed to
```coffeescript
if o.level > LEVEL_LIST or o.level <= LEVEL_LIST and isValue and @variable.base instanceof Obj and not @nestedLhs and not (@param is yes)
```

or even to
```coffeescript
if o.level > LEVEL_LIST or isValue and @variable.base instanceof Obj and not @nestedLhs and not (@param is yes)
```

As far as I understood values of the levels, they're used to control precedence of the compiled code fragments.
From the `nodes.coffee`
```coffeescript
# Levels indicate a node's position in the AST. Useful for knowing if
# parens are necessary or superfluous.
LEVEL_TOP    = 1  # ...;
LEVEL_PAREN  = 2  # (...)
LEVEL_LIST   = 3  # [...]
LEVEL_COND   = 4  # ... ? x : y
LEVEL_OP     = 5  # !...
LEVEL_ACCESS = 6  # ...[0]
```

I guess, it can be confusing  sometimes since level with value 1 (`LEVEL_TOP`) has higher precedence then 2, i.e. `1 > 2` .

Reply:
Okay, so let’s wrap this up. @zdenko and @vendethiel, what do you think the best check is?

And @zdenko you should probably add a comment explaining the conditional, similar to what you’ve written here.

Reply:
I'd say `if o.level > LEVEL_LIST or isValue and @variable.base instanceof Obj and not @nestedLhs and not @param`

Reply:
Agree. It's shorter and IMHO, somehow less confusing.